doc/GUIDE.md

```diff
@@ -0,0 +1,822 @@
+
+## Running Jextract
+
+A native library typically has an `include` directory which contains all the header files
+that define the interface of the library, with one 'main' header file. Let's say we have a
+library called `mylib` stored at `/path/to/mylib` that has a directory `/path/to/mylib/include`
+with all the header files. And let's say that we have a shell open in the root directory
+of the Java project we're working on, which has an `src` source directory corresponding to
+the root package. A typical way to run jextract would be like this:
+
+```sh
+$ jextract \
+  --include-dir /path/to/mylib/include \
+  --output src \
+  --target-package org.jextract.mylib \
+  --library mylib \
+  /path/to/mylib/include/mylib.h
+```
+
+In this command:
+
+- `/path/to/mylib/include/mylib.h` is the main header file of the native library we want
+  to generate bindings for.
+- `--include-dir /path/to/mylib/include` specifies a header file search directory, which
+   is used to find header files included through `#include` in the main header file.
+- `--output src` specifies the root directory for the output. This matches to root package
+  of the project's source directory.
+- `--target-package org.jextract.mylib` specifies the target package to which the generated
+  classes and interfaces will belong. (note that jextract will automatically create the
+  directories representing the package structure under the `src` directory specified
+  through `--output`)
+- `--library mylib` tells jextract that the generated bindings should load the library
+  called `mylib`. (The section on [library loading](#library-loading) discusses how is done)
+
+Besides these options, it is also possible to filter the output of jextract using one of the `--include-XXX` options
+that jextract has. See the section on [filtering](#filtering) for a more detailed overview. See also the full
+list of command line options [here](#command-line-option-reference).
+
+Jextract assumes that the version of a native library that a project uses is relatively stable.
+Therefore, jextract is intended to be run once, and then for the generated sources to be added to the project.
+Jextract only needs to be run again when the native library, or jextract itself are updated. (This is also
+the workflow that jextract itself uses for talking to the libclang library).
+
+### Library Loading
+
+When using the `--library <libspec>` option, the generated code internally uses [`SymbolLookup::libraryLookup`](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/foreign/SymbolLookup.html#libraryLookup(java.nio.file.Path,java.lang.foreign.Arena))
+to load libraries specified by `<libspec>`, after potentially mapping the name of the library to a platform dependent name using [`System::mapLibraryName`](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/System.html#mapLibraryName(java.lang.String)).
```

Comment:
potentially? Are there cases where mapping a library name is a no-op?

Reply:
The `<libspec>` might be a path (using `:`), in which case we don't  do any mapping, right? I don't mind removing the `potentially` here though.

Reply:
In that case, I think it would be clearer if we broke up the sentence. E.g. 

```
When using the --library <libspec> option, the generated code internally uses SymbolLookup::libraryLookup
to load libraries specified by `<libspec>`. If `<libspec>` denotes a library name, the name is then mapped to a platform dependent name using System::mapLibraryName.
```